scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,401 @@
+        for ib_interface in self._fabric_interfaces:
+            ib_interface_to_validate = ib_interface
+            if not ib_interface.startswith("mlx"):
+                ib_interface_to_validate = ib_to_ml_map.get(ib_interface, ib_interface)
+            if ib_interface_to_validate not in ib_interfaces_status:
+                print(f"{ib_interface} is not in the list of IB interfaces")
+                result = False
+            elif not self._check_ib_interface(
+                ib_interface_to_validate, ib_interfaces_status
+            ):
+                print(
+                    f"IB interface {ib_interface} is not active "
+                    f"{ib_interfaces_status[ib_interface]}"
+                )
+                result = False
+        if result:
+            print("All given IB interfaces are active")
+        return result
+
+    @staticmethod
+    def _get_ib_to_ml_map():
+        ret_code, stdout = StandbyNodeHealthChecker._run_command("ibdev2netdev")
+        if ret_code != 0:
+            print("Failed to run ibdev2netdev")
+            return {}
+        line_regex = re.compile(r"^([\w\d_]+) port \d ==> ([\w\d]+)")
+        lines = stdout.splitlines()
+        ib_to_mp = {}
+        for line in lines:
+            cur_line_match = line_regex.match(line)
+            if cur_line_match:
+                mlx_port = cur_line_match.group(1)
+                ib_port = cur_line_match.group(2)
+                ib_to_mp[ib_port] = mlx_port
+        return ib_to_mp
+
+    @staticmethod
+    def _check_ib_interface(ib_interface: str, ib_interfaces_status: dict):
+        if ib_interface in ib_interfaces_status:
+            ib_state = ib_interfaces_status[ib_interface]["State"]
+            physical_state = ib_interfaces_status[ib_interface]["Physical_state"]
+            if ib_state == "active" and physical_state == "linkup":
+                return True
+        return False
+
+    @staticmethod
+    def _run_and_parse_ibstat():
+        retcode, stdout = StandbyNodeHealthChecker._run_command("ibstat")
+        if retcode != 0:
+            print("Cannot run ibstat")
+            return {}
+        lines = stdout.splitlines()
+        ca_info = {}
+        current_ca = None
+
+        for line in lines:
+            line = line.strip()
+            if line.startswith("CA '"):
+                current_ca = line.split("'")[1]
+                ca_info[current_ca] = {}
+            elif current_ca and line.startswith("State:"):
+                ca_info[current_ca]["State"] = line.split(":")[1].strip().lower()
+            elif current_ca and line.startswith("Physical state:"):
+                ca_info[current_ca]["Physical_state"] = (
+                    line.split(":")[1].strip().lower()
+                )
+
+        return ca_info
+
+    @staticmethod
+    def _run_and_parse_ip_link_show():
+        ret_code, ip_link_show_output = StandbyNodeHealthChecker._run_command(
+            "ip link show"
```

Comment:
you can run `ip --json link show` parsing will be much easier

Reply:
Great suggestion, changed!